test/unit/Cardano/Wallet/Primitive/SigningSpec.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Wallet.Primitive.SigningSpec where
+
+import Prelude
+
+import Cardano.Environment
+    ( Network (..), network )
+import Cardano.Wallet.Binary
+    ( decodeSignedTx, encodeSignedTx, toByteString )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Depth (..)
+    , Key
+    , Passphrase (..)
+    , XPrv
+    , keyToAddress
+    , publicKey
+    , unsafeGenerateKeyFromSeed
+    )
+import Cardano.Wallet.Primitive.AddressDiscovery
+    ( AddressScheme (..) )
+import Cardano.Wallet.Primitive.Signing
+    ( mkStdTx )
+import Cardano.Wallet.Primitive.Types
+    ( Address (..)
+    , Coin (..)
+    , Hash (..)
+    , Tx
+    , TxIn (..)
+    , TxOut (..)
+    , TxWitness
+    )
+import Data.Map
+    ( Map )
+import Data.Text.Class
+    ( toText )
+import Test.Hspec
+    ( Spec, describe, it, shouldBe )
+
+import qualified Codec.CBOR.Read as CBOR
+import qualified Data.ByteArray as BA
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.Map as Map
+import qualified Data.Text as T
+
+signedTx :: (Tx, [TxWitness])
+signedTx =
+    let
+        seed = Passphrase $ BA.convert $ BS.pack $ replicate 128 0
+        pass = Passphrase mempty
+        addrXPrv = unsafeGenerateKeyFromSeed (seed, mempty) pass
+        addr = keyToAddress $ publicKey addrXPrv
+        h = Hash "\bztb\249\223n5\189A\239Y&\243tgv\253\223\140\222i\236O\221\&8\205\218>\234\240\238"
+
+        ownedIns = (TxIn h 0, TxOut addr (Coin 1) )
+        outs = [TxOut addr (Coin 1)]
+
+        m = (Map.singleton addr addrXPrv) :: Map Address (Key 'AddressK XPrv)
+
+        fromR = either undefined Prelude.id
+    in
+        fromR $ mkStdTx m (error "no root key", pass) [ownedIns] outs
+
+-- | Reads different fixtures depending on the network.
+--
+-- Fixtures were generated using 'wallet/server/Main.hs' on the
+-- 'anviking/wallet-tx-golden' branch on cardano-sl.
+readFixture :: String -> IO BS.ByteString
+readFixture name
+    = BS.readFile
+        $ "test/data/Cardano/Wallet/Primitive/Signing/"
+          ++ T.unpack (toText network') ++ "/" ++ name
+  where
+    network' =
+        case network of
+            Mainnet -> Mainnet
+            Staging -> Mainnet
+            Testnet -> Testnet
+            Local -> Testnet
```

Comment:
If I recall correctly, the witness includes the protocol magic too, no ? So all tx should be slightly different even between staging and mainnet. Also, for consistency, we're probably better off with having the same file for local and testnet (not sure there's actually any value checking this for the local network...)